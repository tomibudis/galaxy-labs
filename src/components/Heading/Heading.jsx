import React from 'react';
import PropTypes from 'prop-types';

export const TESTID = 'heading-testid';

const Heading = ({ level, className, children, ...restProps }) => {
  return React.createElement(
    `h${level}`,
    { className, 'data-testid': TESTID, ...restProps },
    children
  );
};

Heading.propTypes = {
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  className: PropTypes.string,
  children: PropTypes.node,
};

Heading.defaultProps = {
  level: 4,
};

export default Heading;
