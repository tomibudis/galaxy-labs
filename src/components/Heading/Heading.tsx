import React from 'react';
import PropTypes from 'prop-types';

import { HeadingProps } from '#interface/index';

export const TESTID = 'heading-testid';

const Heading: React.FC<HeadingProps> = ({
  level,
  className,
  children,
  ...restProps
}): JSX.Element => {
  return React.createElement(
    `h${level}`,
    { className, 'data-testid': TESTID, ...restProps },
    children
  );
};

Heading.propTypes = {
  /** Size of font heading */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  /** Additional classname */
  className: PropTypes.string,
  /** React Node */
  children: PropTypes.node,
};

Heading.defaultProps = {
  level: 4,
};

export default Heading;
