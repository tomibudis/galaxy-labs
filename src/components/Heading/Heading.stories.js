import React from 'react';
import Heading from './Heading';

export default {
  title: 'Components/Heading',
  component: Heading,
  argTypes: {
    children: {
      table: {
        disable: true,
      },
    },
  },
};

const Template = (args) => {
  return <Heading {...args}>{args.children}</Heading>;
};

export const TyphographyHeading = Template.bind({});

TyphographyHeading.args = {
  children: 'Typhography h4',
};
