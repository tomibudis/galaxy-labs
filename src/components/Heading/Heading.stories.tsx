import React from 'react';

import { Meta, ComponentStory } from '@storybook/react';

import Heading from './Heading';

export default {
  title: 'Components/Heading',
  component: Heading,
  argTypes: {
    children: {
      table: {
        disable: true,
      },
    },
  },
} as Meta;

const Template: ComponentStory<typeof Heading> = (args) => {
  return <Heading {...args}>{args.children}</Heading>;
};

export const TyphographyHeading = Template.bind({});
TyphographyHeading.args = {
  children: 'Typhography h4',
};
