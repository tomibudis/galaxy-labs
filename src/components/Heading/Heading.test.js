import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Heading, { TESTID } from './Heading';

afterEach(() => {
  cleanup();
});

const HeadingComponent = (props) => {
  return <Heading {...props}>{props?.children || 'Hello Heading..'}</Heading>;
};

describe('<Button />', () => {
  it('render component', () => {
    const { getByTestId } = render(<HeadingComponent />);
    expect(getByTestId(TESTID)).toBeTruthy();
  });

  it('render children', () => {
    const expectedHTML = 'hello world';
    const { getByTestId } = render(
      <HeadingComponent>{expectedHTML}</HeadingComponent>
    );
    expect(getByTestId(TESTID)).toContainHTML(expectedHTML);
  });

  test.each([
    [1, 'h1'],
    [2, 'h2'],
    [3, 'h3'],
    [4, 'h4'],
    [5, 'h5'],
    [6, 'h6'],
  ])('render size level %s of heading', (level, expected) => {
    const { baseElement } = render(<HeadingComponent level={level} />);
    expect(baseElement.querySelector(expected)).toBeTruthy();
  });

  it('matched with snapshot file', () => {
    const { baseElement } = render(<HeadingComponent />);
    expect(baseElement).toMatchSnapshot();
  });
});
