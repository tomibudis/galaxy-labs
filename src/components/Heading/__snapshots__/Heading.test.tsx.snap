// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matched with snapshot file 1`] = `
<body>
  <div>
    <h4
      data-testid="heading-testid"
    >
      Hello Heading..
    </h4>
  </div>
</body>
`;
